src/ngraph/pattern/matcher.cpp

```diff
@@ -243,5 +285,31 @@ namespace ngraph
             }
             return is_match;
         }
+
+        bool Matcher::match(const std::shared_ptr<Node>& graph_node,
+                            const PatternMap& previous_matches)
+        {
+            //clear our state
+            m_match_root.reset();
+            m_pattern_map.clear();
+
+            //insert previous matches
+            m_pattern_map.insert(previous_matches.cbegin(), previous_matches.cend());
+
+            if (!m_pattern_node || !graph_node)
+            {
+                throw "m_pattern_node or graph_node are not set!";
```

Comment:
Someone is asking to be frog for a month.

Reply:
Loooooool :-) Will fix :-)

Reply:
@diyessi actually in my defense, I copied this from the original `match` which was written before we started using `ngraph_error` . Fixed both.